packages/discovery-react-components/src/components/StructuredQuery/components/RuleRow/RuleRow.tsx

```diff
@@ -45,15 +54,20 @@ export const RuleRow: FC<RuleRowProps> = ({
   const showRemoveRuleRowButton =
     structuredQuerySelection.groups[groupId].rows.length > 1 || !isTopLevelGroup;
 
+  // TODO Check out missing prop warning on <ComboBox />
   return (
     <div className={structuredQueryRulesClass} data-testid={`rule-row-${groupId}`}>
       <ComboBox
         id={`structured-query-rules-field-${groupId}`}
-        // TODO: Items is empty for now as it's a required field and retrieving fields for the dropdown
-        // and adding them as items will be addressed in a future issue
-        items={[]}
-        placeholder={messages.fieldDropdownPlaceholderText}
+        items={projectFields}
+        placeholder={
+          haveProjectFields
```

Comment:
i think the check should be `isLoading` not `haveProjectFields` as we don't know whether or not fields will ever be fetched (kind of a false indicator)

Reply:
After some more thought, this does sound like a better strategy and I've made the corresponding changes